### ISSUE TITLE
Concise-arrow bodies lose their return value, so the component or hook disappears

### ISSUE THREAD

**Issue by rboudrouss** [rboudrouss/reactant-analyzer#5]
**Type**: under-approximation. The component or hook disappears entirely.

## Repro

```js
const Concise = ({ label }) => <button>{label}</button>
```

is not reported as a component even under `--show-clean`.

```js
const useConcise = (a) => useState(a)
```

is unresolvable; its consumer reports only
`analysis-limit: hook <hook:2> not found in registry (FN possible)` while its `infinite-loop` goes
unfound (exit 0).

## Cause

The body lowers to an `ExprStmt` with a sealed `Return(Lit(Prim::Unit))` behind it, and `Candidate`
does not carry the arrow's `expression` flag, so none of the three lowerers can dispatch to an
expression-body lowering.

## Secondary, same output

The Info renders `<hook:2>` instead of `useConcise`, which makes it far less actionable.

*Locations: `lowering/cfg_builder.rs`, `Candidate` in `lowering/mod.rs`.*

**Comment by rboudrouss** [comment]
Fixed in `30a00c5`, together with #4 — they could not be separated.

## The fix

`Candidate` did not carry the arrow's `expression` flag, so all three lowerers took the statement path and a concise body returned unit. The flag now travels, and `Candidate::build_cfg` is the single place the dispatch happens, so a new consumer of `Candidate` cannot drop it again — which is exactly how it was lost.

`build_expr_fn_body_cfg` had existed the whole time. Nothing could reach it.

## What it recovers

Concise-arrow utilities whose return value is fresh every call — an extremely common shape — were entirely invisible:

```tsx
const makeConfig = (id) => ({ id, retries: 3 });
const makeHandler = () => () => console.log("hi");
// used as deps → always-unstable-deps, and the whole file read ✓ before
```

## Why it shipped with #4

Fixing this alone is a **soundness regression**, and it nearly went out that way. A concise body now lowers to `Terminator::Return`, and `extract_hooks` only walks `block.stmts` — #4's blind spot. On `const useLocal = (x) => useMystery(x)`:

| | before | after #5 alone |
|---|---|---|
| `analysis-limit` | emitted | **gone** |
| assurances | 4 suspended | **4 issued** |

An honest "I cannot tell" became four verified-clean claims. So #4's terminator hoist had to land first.

## On the two repros in this issue

- **`const useConcise = (a) => useState(a)`** — fixed, the hook resolves and the `analysis-limit` is gone.
- **`const Concise = ({label}) => <button>{label}</button>` not listed under `--show-clean`** — this turned out **not** to be a lowering bug. `Concise` *is* detected, lowered and analysed (the symbol graph and `components_analyzed` both show it). It is not printed because the human renderer skips components with no hooks and no findings, even under `--show-clean`. Cosmetic, and arguably correct — printing every presentational component would flood the output. Say the word if you want `--show-clean` to mean literally all of them.
- **The secondary `<hook:2>` rendering** — no longer reproducible; the hook resolves, so the Info is gone. The name fallback is still there in `analysis_limit_info.rs` for genuinely unnamed labels, and I could not construct a case that hits it.

## Measurement

Corpus 1314 → 1340 (+26: 10 removed, 36 added), both halves together. The 10 removals were read to source and are true FPs. Details and the untriaged-additions caveat in `docs/precision-log.md`.

One residual, tracked as #140: a hook reached through a return has no line number, because `Terminator::Return` carries no span.
